mmv1/third_party/terraform/acctest/bootstrap_test_utils.go.erb

```diff
@@ -0,0 +1,1556 @@
+	}
+	return storagePool.SelfLink
+}
+
+func SetupProjectsAndGetAccessToken(org, billing, pid, service string, config *transport_tpg.Config) (string, error) {
+	// Create project-1 and project-2
+	rmService := config.NewResourceManagerClient(config.UserAgent)
+
+	project := &cloudresourcemanager.Project{
+		ProjectId: pid,
+		Name:      pid,
+		Parent: &cloudresourcemanager.ResourceId{
+			Id:   org,
+			Type: "organization",
+		},
+	}
+
+	var op *cloudresourcemanager.Operation
+	err := transport_tpg.Retry(transport_tpg.RetryOptions{
+		RetryFunc: func() (reqErr error) {
+			op, reqErr = rmService.Projects.Create(project).Do()
+			return reqErr
+		},
+		Timeout: 5 * time.Minute,
+	})
+	if err != nil {
+		return "", err
+	}
+
+	// Wait for the operation to complete
+	opAsMap, err := tpgresource.ConvertToMap(op)
+	if err != nil {
+		return "", err
+	}
+
+	waitErr := resourcemanager.ResourceManagerOperationWaitTime(config, opAsMap, "creating project", config.UserAgent, 5*time.Minute)
+	if waitErr != nil {
+		return "", waitErr
+	}
+
+	ba := &cloudbilling.ProjectBillingInfo{
+		BillingAccountName: fmt.Sprintf("billingAccounts/%s", billing),
+	}
+	_, err = config.NewBillingClient(config.UserAgent).Projects.UpdateBillingInfo(resourcemanager.PrefixedProject(pid), ba).Do()
+	if err != nil {
+		return "", err
+	}
+
+	p2 := fmt.Sprintf("%s-2", pid)
+	project.ProjectId = p2
+	project.Name = fmt.Sprintf("%s-2", pid)
+
+	err = transport_tpg.Retry(transport_tpg.RetryOptions{
+		RetryFunc: func() (reqErr error) {
+			op, reqErr = rmService.Projects.Create(project).Do()
+			return reqErr
+		},
+		Timeout: 5 * time.Minute,
+	})
+	if err != nil {
+		return "", err
+	}
+
+	// Wait for the operation to complete
+	opAsMap, err = tpgresource.ConvertToMap(op)
+	if err != nil {
+		return "", err
+	}
+
+	waitErr = resourcemanager.ResourceManagerOperationWaitTime(config, opAsMap, "creating project", config.UserAgent, 5*time.Minute)
+	if waitErr != nil {
+		return "", waitErr
+	}
+
+	_, err = config.NewBillingClient(config.UserAgent).Projects.UpdateBillingInfo(resourcemanager.PrefixedProject(p2), ba).Do()
+	if err != nil {
+		return "", err
+	}
+
+	// Enable the appropriate service in project-2 only
+	suService := config.NewServiceUsageClient(config.UserAgent)
+
+	serviceReq := &serviceusage.BatchEnableServicesRequest{
+		ServiceIds: []string{fmt.Sprintf("%s.googleapis.com", service)},
+	}
+
+	_, err = suService.Services.BatchEnable(fmt.Sprintf("projects/%s", p2), serviceReq).Do()
+	if err != nil {
+		return "", err
+	}
+	time.Sleep(1 * time.Minute)
+
+<% unless version == 'ga' -%>
+	// For Firebase test only
+	// After project-2 is created and the firebase API is enabled, 
+	// we need to make the GCP project a Firebase project.
+	// Using google_firebase_project in the acceptance test wasn't sufficient.
+	if service == "firebase" {
+		url := fmt.Sprintf("%sprojects/%s:addFirebase", config.FirebaseBasePath, p2)
```

Comment:
It may be better to handle this by adding more permissions ([see docs](https://firebase.google.com/docs/reference/firebase-management/rest/v1beta1/projects/addFirebase)) down near L1327

Reply:
This would require giving the service account:
- roles/serviceusage.serviceUsageAdmin
- A role that includes permission resourcemanager.projects.get